js/bruh-dash.js

```diff
@@ -10,89 +10,215 @@ var global = window || GLOBAL;
 global.bruhdash = {
 
   // returns the first element of an array
-  first: function () {
-      
+  first: function (array) {
+    return array[0];
   },
 
-  // returns the last element of an array
-  last: function () {
 
+  // returns the last element of an array
+  last: function (array) {
+    return array[array.length-1];
   },
 
   // returns the index of the first matching element from left to right
-  indexOf: function () {
+  indexOf: function (array, element) {
+    for (var i=0;i<array.length;i++){
+      if(array[i] === element){
+        console.log(array[i]);
+        console.log(i);
+        return i;
+    }
 
+  }
+  return -1;
   },
 
-  // returns the index of the first matching element from left to right
-  lastIndexof: function () {
-
+  // returns the index of the first matching element from right to left
+  lastIndexOf: function (array, element) {
+    for (var i = array.length-1; i >= 0; i--){
+      if(array[i] === element){
+        console.log(array[i]);
+        console.log(i);
+        return i;
+      }
+    }
+    return -1;
   },
 
   // returns an array with all elements except for the last element
-  initial: function () {
+  initial: function (array) {
+    return array.slice(0, array.length-1);
+
 
   },
-  
-  // returns an array with all falsey values removed
-  compact: function() {
 
+  // returns an array with all falsey values removed
+  compact: function(array) {
+    return array.filter(Boolean);
   },
 
   // creates a slice of an array from the start index up to but not including the end index
-  slice: function () {
+  slice: function (array, start, end) {
+    var newArray = array.slice(start, end);
+    console.log(newArray);
+    return newArray;
 
   },
 
   // returns a slice of array with n elements dropped from the beignning
-  drop: function(){
-
+  drop: function(array, amount){
```

Comment:
this can be simplified. Your `else if` fill return the first value in the array. I would think if someone passed in 'a' you would just return the array.
Remove console.log